client/packages/common/src/hooks/useInterval/useInterval.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useRef, useInsertionEffect, useCallback } from 'react';
+
+type ArgsType = (number | string | object)[]; // Avoid `any` type
+export type Callback = () => void;
+export type EventCallback = (...args: ArgsType) => void;
+export type Delay = number | null;
+
+/**
+ * Hook that lets you extract non-reactive logic into an effect event.
+ * This is a kind of polyfill for the `useEffectEvent` hook that is still
+ * in the experimental phase. Once it's stable, we can replace this hook with it.
+ * @param callback An event callback function
+ * @returns A function that can be called to execute the callback
+ */
+const useEffectEvent = (callback: EventCallback) => {
```

Comment:
Thanks for the explanation. @mark-prins this looks very useful for other parts of our code where the linter gives a warning, but you don't want something like this in your dependency array...